pkg/controller/apimanagerbackup/apimanagerbackup_logic_reconciler.go

```diff
@@ -0,0 +1,398 @@
+		// next step, which would fail because Job is not deleted nor the PVC associated
+		// to it, which would provoke an error when trying to reuse it for the next job PVC
+		err = r.GetResource(common.ObjectKey(desired), existing)
+		if err != nil && !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+
+		substepFinished := true
+		r.cr.Status.SecretsAndConfigMapsCleanupSubStepFinished = &substepFinished
+		err = r.UpdateResourceStatus(r.cr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.Logger().Info("Job cleant up successfully. Requeuing", "Job Name", desired.Name)
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *APIManagerBackupLogicReconciler) reconcileAPIManagerCustomResourceBackupToPVCJob() (reconcile.Result, error) {
+	desired := r.apiManagerBackup.BackupAPIManagerCustomResourceToPVCJob()
+	if desired == nil {
+		return reconcile.Result{}, nil
+	}
+
+	// TODO create Mutator function for Jobs??
+	existing := &batchv1.Job{}
+	// Check if this step has finished
+	if r.cr.APIManagerCustomResourceBackupStepFinished() {
+		return reconcile.Result{}, nil
+	}
+
+	// Check if backup substep has completed
+	if !r.cr.APIManagerCustomResourceBackupSubStepFinished() {
+		err := r.ReconcileResource(existing, desired, reconcilers.CreateOnlyMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		substepFinished := true
+		if existing.Status.Succeeded != *desired.Spec.Completions {
+			r.Logger().Info("Job has still not finished", "Job Name", desired.Name, "Actively running Pods", existing.Status.Active, "Failed pods", existing.Status.Failed)
+			return reconcile.Result{Requeue: true}, nil
+		}
+
+		r.cr.Status.APIManagerCustomResourceBackupSubStepFinished = &substepFinished
+		err = r.UpdateResourceStatus(r.cr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.Logger().Info("Job finished successfully. Requeing", "Job Name", desired.Name)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Check if cleanup substep has completed
+	if !r.cr.APIManagerCustomResourceCleanupSubStepFinished() {
+		common.TagToObjectDeleteWithPropagationPolicy(desired, metav1.DeletePropagationForeground)
+		err := r.ReconcileResource(existing, desired, reconcilers.CreateOnlyMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		err = r.GetResource(common.ObjectKey(desired), existing)
+		if err != nil && !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+
+		substepFinished := true
+		r.cr.Status.APIManagerCustomResourceCleanupSubStepFinished = &substepFinished
+		err = r.UpdateResourceStatus(r.cr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.Logger().Info("Job cleant up successfully. Requeuing", "Job Name", desired.Name)
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *APIManagerBackupLogicReconciler) reconcileBackupSystemFileStoragePVCToPVCJob() (reconcile.Result, error) {
```

Comment:
Same sort of comments as for reconcileBackupSecretsAndConfigMapsToPVCJob

Reply:
I understand you refer to your comment over here #358 (comment). I answered there. If you were not referring to that please tell me

Reply:
This is obsolete as the state check was refactored following another approach. Marking as resolved.